docs/cookbook/custom-email.rst

```diff
@@ -0,0 +1,176 @@
+How to send a custom e-mail?
+============================
+
+.. note::
+
+    This cookbook is suitable for a clean :doc:`sylius-standard installation </book/installation>`.
+    For more general tips, while using :doc:`SyliusMailerBundle </bundles/SyliusMailerBundle/index>`
+    go to `Sending configurable e-mails in Symfony Blogpost <http://sylius.org/blog/sending-configurable-e-mails-in-symfony>`_.
+
+Currently **Sylius** is sending e-mails only in a few "must-have" cases - see :doc:`E-mails documentation </book/emails>`.
+Of course these cases may not be sufficient for your business needs, if so you will need to create your own, custom e-mails inside your system.
+
+On a basic example we will now teach how to do it.
+
+Let's assume that you would like such a feature in your system:
+
+.. code-block:: text
+
+    Feature: Sending a notification email to the administrator when a product is out of stock
+        In order to be aware which products become out of stock
+        As an Administrator
+        I want to be notified via email when products become out of stock
+
+To achieve that you will need to:
+
+1. Create a new e-mail that will be sent:
+-----------------------------------------
+
+* prepare a template for your email in the ``AppBundle/Resources/views/Email``.
+
+.. code-block:: twig
+
+    {# AppBundle/Resources/views/Email/out_of_stock.html.twig #}
+    {% block subject %}
+        One of your products has become out of stock.
+    {% endblock %}
+
+    {% block body %}
+        {% autoescape %}
+            The {{ variant.name }} variant is out of stock!
+        {% endautoescape %}
+    {% endblock %}
+
+* configure the email under ``sylius_mailer:`` in the ``app/config/emails.yml`` included in ``app/config/yml``.
+
+.. code-block:: yaml
+
+    # app/config/emails.yml
+    sylius_mailer:
+        sender:
+            name: Example.com
+            address: no-reply@example.com
+        templates:
+            app.email.order_confirmation.name: "AppBundle:Email:out_of_stock.html.twig"
+        emails:
+            out_of_stock:
+                subject: "A product has become out of stock!"
+                template: "AppBundle:Email:out_of_stock.html.twig"
+
+.. code-block:: yaml
+
+    # app/config/config.yml
+    imports:
+        - { resource: "emails.yml" }
+
+2. Create a new Email Manager:
+------------------------------
+
+* Create the class;
```

Comment:
Is this point necessary? The section could just be called "Create an Email Manager class"